client/app/core/navigation/navigation-controller.js

```diff
@@ -82,11 +82,26 @@ export function NavigationController (Text, Navigation, Session, API_BASE, Shopp
     })
     vm.items = Navigation.init()
     refresh()
+    setMenuItemActive()
     if (ShoppingCart.allowed()) {
       ShoppingCart.reload()
     }
+    $scope.$on('$stateChangeSuccess', () => {
+      setMenuItemActive()
+    })
+  }
+  function setMenuItemActive () {
```

Comment:
What about putting the logic of this function in the `doCheck` rather than a timeout?